ui-modules/blueprint-composer/app/components/util/model/entity.model.js

```diff
@@ -281,6 +282,13 @@ export class Entity {
      */
     removeLocation() {
         LOCATIONS.delete(this);
+        this.miscData.delete('locationName');
+        this.miscData.delete('locationIcon');
+        
+        // this field provides a way for consumers to detect if the location was explicitly removed;
+        // this can be useful to prevent default locations from being applied
+        this.miscData.set('locationRemoved', true);
```

Comment:
I don't get what this is for, even with the comment. Can you explain the use-case please?

Reply:
in a downstream project we automatically add a location if none is present.  however if the user removes it in the UI, we in that downstream code want to now that we are able _not_ to re-add it in that case.